Description: Fix return type of strlen calls and
 snprintf/prot_printf format strings.
 Most of the original patch has been dropped (since already fixed
 upstream), but kept the "manual" address format to %p change.
 Might make sense to just upstream it, especially since the
 hardcoded format string is a bit susceptible to pointer width
 issues.
Author: Sven Mueller <debian@incase.de>
Forwarded: no
Last-Update: 2022-05-05

---
 imtest/imtest.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/imtest/imtest.c b/imtest/imtest.c
index a6fcf91e6..962b14bdf 100644
--- a/imtest/imtest.c
+++ b/imtest/imtest.c
@@ -637,14 +637,14 @@ static long bio_dump_cb(BIO * bio, int cmd, const char *argp, int argi,
         return (ret);
 
     if (cmd == (BIO_CB_READ | BIO_CB_RETURN)) {
-        printf("read from %08lX [%08lX] (%d bytes => %ld (0x%lX))\n",
-               (unsigned long) bio, (unsigned long) argp,
+        printf("read from %p [%p] (%d bytes => %ld (0x%lX))\n",
+               (void*) bio, (void*) argp,
                argi, ret, ret);
         tls_dump(argp, (int) ret);
         return (ret);
     } else if (cmd == (BIO_CB_WRITE | BIO_CB_RETURN)) {
-        printf("write to %08lX [%08lX] (%d bytes => %ld (0x%lX))\n",
-               (unsigned long) bio, (unsigned long) argp,
+        printf("write to %p [%p] (%d bytes => %ld (0x%lX))\n",
+               (void*) bio, (void*) argp,
                argi, ret, ret);
         tls_dump(argp, (int) ret);
     }
